js/main.js

```diff
@@ -630,16 +656,270 @@ function showCluster(a) {
+                a.push('<a href="#' + currentElem + '" onclick="findNodesForAttributes(\'' + attributeName + "'," + currentElem + "," + !1 + "," + results + ")\">" + currentElem + "</a>");
+            }
+        } else
+            a.push("<i>No results found.</i>");
+
+        var results = $(".results");
+        results.html(a.join(""));
+        results.show();
+        alert("Let's see if results are there");
+    }
+
+    // Otherwise, we're doing our continuous variable search
+    else {
+        sigInst.iterNodes(function (a) {
+            var currentValue = Number(a.attr.attributes[attributeName]);
+            // Otherwise, we have a minimum and maximum value to consider for our search    
+            if(currentValue >= Number(minValue) && currentValue <= Number(maxValue))
+                outputList.push(a.id);
+        });
+    }
+
+    showClusterForAttribute(attributeName, outputList);
+}
+
+// Show cluster given the attribute "a" that you're looking for
+function showClusterForAttribute(a, b) {
+    if (b && 0 < b.length) {
+        showGroups(!1);
+        sigInst.detail = !0;
+        b.sort();
+        sigInst.iterEdges(function (a) {
+            a.hidden = !1;
+            a.attr.lineWidth = !1;
+            a.attr.color = !1
+        });
+        sigInst.iterNodes(function (a) {
+            a.hidden = !0
+        });
+        for (var f = [], e = [], c = 0, g = b.length; c < g; c++) {
+            //alert("b is " + b + " and c is " + c);
+            var d = sigInst._core.graph.nodesIndex[b[c]];
+            !0 == d.hidden && (e.push(b[c]), d.hidden = !1, d.attr.lineWidth = !1, d.attr.color = d.color, f.push('<li class="membership"><a href="#'+d.label+'" onmouseover="sigInst._core.plotter.drawHoverNode(sigInst._core.graph.nodesIndex[\'' + d.id + "'])\" onclick=\"nodeActive('" + d.id + '\')" onmouseout="sigInst.refresh()">' + d.label + "</a></li>"))
+        }
+        sigInst.clusters[a] = e;
+        sigInst.draw(2, 2, 2, 2);
+        // The next line needs to be changed so that we have a good description of the attribute we're searching for
+        // $GP.info_name.html("<b>" + sigInst.clusterNames[a] + "</b>");
+        $GP.info_name.html("<b>" + a + "</b>");
+        $GP.info_data.hide();
+        $GP.info_p.html("Group Members:");
+        $GP.info_link.find("ul").html(f.join(""));
+        $GP.info.animate({width:'show'},350);
+        //$GP.search.clean();
+        $GP.cluster.hide();
+        return !0
+    }
+    return !1
+}
+
+
+
+
+function populateSearch() { 
+    // THIS IS GENIE UPDATED
+    var firstElem;
+    for(elem in sigInst._core.graph.nodesIndex){
+        firstElem = elem;
+        break;
+    }
+    if(isNaN(firstElem))
+        firstElem = String(firstElem);
+    var b = sigInst._core.graph.nodesIndex[firstElem];
+    var f = b.attr;
+    
```

Comment:
Declare var f;
and finish everything inside the for loop. Same as previous comment

Reply:
I think this for loop needs to be there, unlike the other ones.

Reply:
Keep the loop i did not mean that. I meant it could be changed to 
var f;
for(elem in sigInst._core.graph.nodesIndex){
   if(isNaN(elem)) {
        elem = String(elem)
   }
   f = sigInst._core.graph.nodesIndex[elem].attr;
}

also check if we need 
 if(isNaN(elem)) {
        elem = String(elem)
   }
i do think it is required

Reply:
Ahh ok I had converted this into a "getFirstElem()" function, so it now looks like this:

var firstElem = getFirstElem();
if(isNaN(firstElem))
         firstElem = String(firstElem);
 var b = sigInst._core.graph.nodesIndex[firstElem];
 var f = b.attr;